src/dvb.c

```diff
@@ -340,6 +340,234 @@ void copy_dvb_parameters(transponder *s, transponder *d) {
+// based on the modulation. The top is the standard OK signal value in dB.
+float get_db_snr_map(transponder *tp) {
+    float db_map;
+    int top = 10; // Default factor 1.0: dB value ==> % (3dB => 3%)
+
+    switch (tp->sys) {
+
+    case SYS_DVBC_ANNEX_AC:
+    case SYS_DVBC_ANNEX_B:
+    case SYS_DVBC_ANNEX_C: /* not sure, to be tested */
+        switch (tp->mtype) {
+
+        case QAM_256:
+            top = DVB_C__QAM_256__FEC_NONE;
+            break;
+
+        case QAM_64:
+            top = DVB_C__QAM_64___FEC_NONE;
+            break;
+
+        default:
+            top = DVB_C__OTHER;
+            LOG("get_db_snr_map -> Mudulation SNR scale mapping DVB-C not tested !!");
+        }
+        break;
+
+    case SYS_DVBS:
+        switch (tp->mtype) {
+
+        case QPSK:
+            switch (tp->fec) {
+
+            case FEC_1_2:
+                top = DVB_S__QPSK__FEC_1_2;
+                break;
+
+            case FEC_2_3:
+                top = DVB_S__QPSK__FEC_2_3;
+                break;
+
+            case FEC_3_4:
+                top = DVB_S__QPSK__FEC_3_4;
+                break;
+
+            case FEC_5_6:
+                top = DVB_S__QPSK__FEC_5_6;
+                break;
+
+            case FEC_7_8:
+                top = DVB_S__QPSK__FEC_7_8;
+                break;
+
+            default:
+                top = DVB_S__OTHER;
+            }
+
+        default:
+            top = DVB_S__OTHER;
+            LOG("get_db_snr_map -> Mudulation SNR scale mapping DVB-S not tested !!");
+        }
+        break;
+
+    case SYS_DVBS2:
+        switch (tp->mtype) {
+
+        case QPSK:
+        switch (tp->fec) {
+
+            case FEC_1_2:
+                top = DVB_S2_QPSK__FEC_1_2;
+                break;
+
+            case FEC_2_3:
+                top = DVB_S2_QPSK__FEC_2_3;
+                break;
+
+            case FEC_3_4:
+                top = DVB_S2_QPSK__FEC_3_4;
+                break;
+
+            case FEC_5_6:
+                top = DVB_S2_QPSK__FEC_5_6;
+                break;
+
+            case FEC_8_9:
+                top = DVB_S2_QPSK__FEC_8_9;
+                break;
+
+            case FEC_9_10:
+                top = DVB_S2_QPSK__FEC_9_10;
+                break;
+
+            default:
+                top = DVB_S2_OTHER;
+            }
+
+        case PSK_8:
+        switch (tp->fec) {
+
+            case FEC_2_3:
+                top = DVB_S2_PSK_8_FEC_2_3;
+                break;
+
+            case FEC_3_4:
+                top = DVB_S2_PSK_8_FEC_3_4;
+                break;
+
+            case FEC_5_6:
+                top = DVB_S2_PSK_8_FEC_5_6;
+                break;
+
+            case FEC_8_9:
+                top = DVB_S2_PSK_8_FEC_8_9;
+                break;
+
+            default:
+                top = DVB_S2_PSK_8_OTHER;
+            }
+
+        default:
+            top = DVB_S2_OTHER;
+            LOG("get_db_snr_map -> Mudulation SNR scale mapping DVB-S2 not tested !!");
```

Comment:
Typo

Reply:
Fixed!

Reply:
Hi @catalinii ,

Improved with DVB-T2 support and with some range checks. Now the default behaviour is to use 80dB as the 100% SNR scale.

Please check it and merge.
Regards.